src/App.tsx

```diff
@@ -1,24 +1,337 @@
+              disabled={isCreating}
+            />
+          </form>
+        </header>
+
+        <section className="todoapp__main">
+          <TransitionGroup>
+            {visibleTodos.map(todo => (
+              <CSSTransition
+                key={todo.id}
+                timeout={300}
+                classNames="item"
+              >
+                <TodoItem
+                  todo={todo}
+                  onDelete={handleDeleteTodo}
+                  isProcessed={processedTodoIds.includes(todo.id)}
+                />
+              </CSSTransition>
+            ))}
+
+            {tempTodo !== null && (
+              <CSSTransition
+                key={0}
+                timeout={300}
+                classNames="temp-item"
+              >
+                <TodoItem
+                  todo={tempTodo}
+                  isProcessed
+                />
+              </CSSTransition>
+            )}
+          </TransitionGroup>
+
+        </section>
+
+        {todos.length !== 0 && (
+          <footer className="todoapp__footer">
+            <span className="todo-count">
+              {`${todos.filter(todo => !todo.completed).length} items left`}
+            </span>
+
+            <nav className="filter">
+              <a
+                href="#/"
+                className={classNames(
+                  'filter__link',
+                  { selected: filterStatus === 'all' },
+                )}
+                onClick={() => setFilterStatus(FilterStatus.all)}
+              >
+                All
+              </a>
+
+              <a
+                href="#/active"
+                className={classNames(
+                  'filter__link',
+                  { selected: filterStatus === 'active' },
+                )}
+                onClick={() => setFilterStatus(FilterStatus.active)}
+              >
+                Active
+              </a>
+
+              <a
+                href="#/completed"
+                className={classNames(
+                  'filter__link',
+                  { selected: filterStatus === 'completed' },
+                )}
+                onClick={() => setFilterStatus(FilterStatus.completed)}
+              >
+                Completed
+              </a>
+            </nav>
+
+            <button
+              type="button"
+              className="todoapp__clear-completed"
+              disabled={todos.every(todo => !todo.completed)}
+              onClick={() => handleClearCompleted()}
+            >
+              Clear completed
+            </button>
+          </footer>
+        )}
+      </div>
+
+      <div className={classNames(
+        'notification',
+        'is-danger',
+        'is-light',
+        'has-text-weight-normal',
+        { hidden: !error.isError },
+      )}
+      >
+        <button
+          type="button"
+          className="delete"
+          onClick={() => dispatch({ type: 'remove_error' })}
```

Comment:
it's better to create a separate handlers instead of passing an anonymous function